model_utilities/train/finetuner.py

```diff
@@ -43,52 +44,83 @@ def __plot_losses(self, training_losses: list[float], validation_losses: list[fl
         plt.legend()
         plt.savefig("./debug/graphs/losses.png")
 
-    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, dataset: ds.Dataset, optimizer: Adafactor,
+    def __tokenize_dataset(self, dataset: ds.Dataset, tokenizer: NllbTokenizerFast, max_length: int) -> Dataset:
+        # Not meant to be called outside of this function
+        def tokenize_fn(dataset: ds.Dataset, src_lang: str, tgt_lang: str, tokenizer: NllbTokenizerFast, max_length: int) -> BatchEncoding:
+            tokenizer.src_lang = src_lang
+            tokenizer.tgt_lang = tgt_lang
+            tokenized_input = tokenizer(dataset[src_lang], text_target=dataset[tgt_lang], return_tensors='pt', padding=True, truncation=True, max_length=max_length)
+
+            # -100 is the default value ignored by pytorch
+            # TODO: Check performance after removing this line
+            tokenized_input.labels[tokenized_input.labels == tokenizer.pad_token_id] = -100
+            return tokenized_input
+
+        tokenized_train_dataset_pl_to_csb = dataset.map(
+            lambda row: tokenize_fn(row, "pol_Latn", "csb_Latn", tokenizer, max_length),
+            batched=True
+        )
+        tokenized_train_dataset_csb_to_pl = dataset.map(
+            lambda row: tokenize_fn(row, "csb_Latn", "pol_Latn", tokenizer, max_length),
+            batched=True
+        )
+
+        tokenized_dataset = ds.interleave_datasets([tokenized_train_dataset_pl_to_csb, tokenized_train_dataset_csb_to_pl])
+        tokenized_dataset = tokenized_dataset.remove_columns(["pol_Latn", "csb_Latn"])
+
+        return tokenized_dataset
+
+    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizerFast, dataset: ds.Dataset, optimizer: Adafactor,
                 config: ConfigParser) -> None:
         self.__log_train_config(config)
 
+        # Prepare variables
```

Comment:
This is an unnecessary comment. Perhaps a log would be better to indicate the process?

Reply:
eh, maybe. i like it though as it provides a visible break in code between the steps.

I agree though that it looks dumb because I stopped adding them midway. will consider logging.